lib/internal/bootstrap/loaders.js

```diff
@@ -140,7 +140,8 @@ let internalBinding;
 const loaderExports = {
   internalBinding,
   NativeModule,
-  require: nativeModuleRequire
+  require: nativeModuleRequire,
+  cjsContext: { __proto__: null, process: undefined, Buffer: undefined }
```

Comment:
Is `__proto__: null` necessary? I assume so, because it’s used as the context object for CJS compilation below, but I feel like it could be responsible for part of the performance impact. If it is necessary, it might also be good to explicitly test the fact that inherited methods aren’t entered into the CJS module’s scope?

Reply:
To my understanding context is equivalent to regular with() so all delegated enumerable properties would be visible, JS language builtins are not enumerable however things matching web specs are. To test, you should ensure that you define an enumerable property. If context is acting like with() it should expose the enumerable properties even if added late in execution.

Reply:
To my understanding context is equivalent to regular with() so all delegated enumerable properties would be visible, JS language builtins are not enumerable however things matching web specs are. To test, you should ensure that you define an enumerable property. If context is acting like with() it should expose the enumerable properties even if added late in execution.